### ISSUE TITLE
FhirXmlNode.Parse() does not parse uppercase xml properties in none fhir xml strings

### ISSUE THREAD

**Issue by webersteffen** [FirelyTeam/firely-net-sdk#1608]
**Is your feature request related to a problem? Please describe.**
One year ago the support for ccda xml parsing was added to the SDK xml parser. It is used at the moment in the mapping plugin for parsing an incomming xml string (in this case ccda) and transform the resulting sourcenode into an itypedelement for the fhir mapper. To be clear, this xml string are not application/fhir+xml strings, the fields are serialized like: <myproperty>myvalue</myproperty>

Recently I also used this route for parsing a none fhir xml we got from a customer and map it, but i ran into a little problem there. The FhirXmlNode.Parse() function did not recognize the xml properties, because the property names were all written in uppercases. Its sees the parser does not recognize this, because primitive values in fhir are normally written in lowercase. But in this case the input is not fhir. 
As far as I understood it, the elementmodel (in this case the ISourceNode) was planned as a fhir independent model. So if there is already the ability in the sdk to parse none fhir xml strings, shouldn´t the parser also be able to parse these strings without the restriction to write primitive value nodes lowercase?

**Describe the solution you'd like**
The parser behind the FhirXmlNode should be able to parse also xml strings that are not according to the fhir standart, but to normal xml standarts. 

**Additional context**
I opend a branch and added two test in that, to show the problme with some test data. The branch can be found  here https://github.com/FirelyTeam/firely-net-sdk/tree/issue/fhirxmlnode-parses-no-uppercase-properties

**Comment by ewoutkramer** [comment]
Yes, the element model is FHIR-independent (or we're trying to), but the same is not true for the FhirXmlNode.  That's about -the name says it- *FHIR* XML. So there are some hard assumptions about the format.  Last year when we did CDA it was just with luck that we managed to combine it in a single parser, but the same will probably not be true for any non-FHIR XML format....

A solution is to copy the source code for the FhirXmlNode and adapt it so it fits the custom XML format.  I don't think that's a lot of differences - I mainly use the first capital letter to figure out the ResourceType - which is probably not relevant in your case.
